File: backend/app/schemas/quick_setup.py
"""快速设置相关 schemas"""

from enum import StrEnum

from pydantic import BaseModel, Field


class ChecklistItemStatus(StrEnum):
    """检查项状态"""

    OK = "ok"  # 已正确配置
    DEFAULT = "default"  # 使用默认值
    MISSING = "missing"  # 缺失
    ERROR = "error"  # 错误


class ChecklistItem(BaseModel):
    """检查项"""

    key: str
    label: str
    category: str
    status: ChecklistItemStatus
    current_value: str | None = None
    default_value: str | None = None
    description: str | None = None
    step_index: int | None = None


class ChecklistResponse(BaseModel):
    """检查清单响应"""

    items: list[ChecklistItem]
    total: int
    ok_count: int
    default_count: int
    missing_count: int


class SetupStepStatus(StrEnum):
    """设置步骤状态"""

    PENDING = "pending"
    COMPLETED = "completed"
    SKIPPED = "skipped"


class SetupStep(BaseModel):
    """设置步骤"""

    index: int
    key: str
    title: str
    description: str | None = None
    status: SetupStepStatus = SetupStepStatus.PENDING
    is_required: bool = True


class QuickSetupState(BaseModel):
    """快速设置状态"""

    current_step: int = 0
    steps: list[SetupStep] = Field(default_factory=list)
    essential_completed: bool = False
    setup_level: str = "none"  # none, essential, full
    data: dict = Field(default_factory=dict)


class EssentialSetupRequest(BaseModel):
    """精简设置请求"""

    llm_provider: str
    llm_api_key: str
    llm_base_url: str
    llm_chat_model: str


class TestConnectionRequest(BaseModel):
    """测试连接请求"""

    llm_provider: str
    llm_api_key: str
    llm_base_url: str
    llm_chat_model: str


class TestConnectionResponse(BaseModel):
    """测试连接响应"""

    success: bool
    message: str
    latency_ms: int | None = None

File: frontend/lib/api/quick-setup.ts
/**
 * 快速设置 API
 */

import { apiGet, apiPost } from "@/lib/api/client";

export interface SetupStep {
  index: number;
  key: string;
  title: string;
  description: string | null;
  status: "pending" | "completed" | "skipped";
  is_required: boolean;
}

export interface QuickSetupState {
  current_step: number;
  steps: SetupStep[];
  essential_completed: boolean;
  setup_level: string;
  data: Record<string, unknown>;
}

export interface ChecklistItem {
  key: string;
  label: string;
  category: string;
  status: "ok" | "default" | "missing" | "error";
  current_value: string | null;
  default_value: string | null;
  description: string | null;
}

export interface ChecklistResponse {
  items: ChecklistItem[];
  total: number;
  ok_count: number;
  default_count: number;
  missing_count: number;
}

export interface EssentialSetupRequest {
  llm_provider: string;
  llm_api_key: string;
  llm_base_url: string;
  llm_chat_model: string;
}

export interface TestConnectionRequest {
  llm_provider: string;
  llm_api_key: string;
  llm_base_url: string;
  llm_chat_model: string;
}

export interface TestConnectionResponse {
  success: boolean;
  message: string;
  latency_ms: number | null;
}

export function getSetupState(): Promise<QuickSetupState> {
  return apiGet<QuickSetupState>("/quick-setup/state");
}

export function getChecklist(): Promise<ChecklistResponse> {
  return apiGet<ChecklistResponse>("/quick-setup/checklist");
}

export function completeEssentialSetup(
  data: EssentialSetupRequest
): Promise<QuickSetupState> {
  return apiPost<QuickSetupState>("/quick-setup/essential", data);
}

export function testConnection(
  data: TestConnectionRequest
): Promise<TestConnectionResponse> {
  return apiPost<TestConnectionResponse>("/quick-setup/test-connection", data);
}

export function resetSetup(): Promise<QuickSetupState> {
  return apiPost<QuickSetupState>("/quick-setup/reset");
}

export interface ModelInfo {
  id: string;
  name: string;
  provider: string;
  tool_calling: boolean;
  structured_output: boolean;
  reasoning: boolean;
  max_input_tokens: number;
  max_output_tokens: number;
}

export interface ProviderInfo {
  id: string;
  name: string;
  base_url: string;
  models: ModelInfo[];
}

export interface ProvidersResponse {
  providers: ProviderInfo[];
}

export interface ProviderModelsResponse {
  provider_id: string;
  base_url: string;
  models: ModelInfo[];
}

export function getProviders(): Promise<ProvidersResponse> {
  return apiGet<ProvidersResponse>("/quick-setup/providers");
}

export function getProviderModels(providerId: string): Promise<ProviderModelsResponse> {
  return apiGet<ProviderModelsResponse>(`/quick-setup/providers/${providerId}/models`);
}
